src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Double128Vector.java

```diff
@@ -864,9 +866,40 @@ public void intoArray(int[] a, int offset) {
                         a[offset + i] = laneSource(i);
                     }
                 }
-            }
+           }
+
         }
 
+        @Override
+        @ForceInline
+        public void intoMemorySegment(MemorySegment ms, long offset, ByteOrder bo) {
+                      switch (length()) {
+                          case 1 -> ms.set(ValueLayout.OfInt.JAVA_INT, offset, laneSource(0));
+                          case 2 -> toBitsVector()
+                                  .convertShape(VectorOperators.L2I, IntVector.SPECIES_64, 0)
+                                  .reinterpretAsInts()
+                                  .intoMemorySegment(ms, offset, bo);
+                          case 4 -> toBitsVector()
+                                  .convertShape(VectorOperators.L2I, IntVector.SPECIES_128, 0)
+                                  .reinterpretAsInts()
+                                  .intoMemorySegment(ms, offset, bo);
+                          case 8 -> toBitsVector()
+                                  .convertShape(VectorOperators.L2I, IntVector.SPECIES_256, 0)
+                                  .reinterpretAsInts()
+                                  .intoMemorySegment(ms, offset, bo);
+                          case 16 -> toBitsVector()
+                                  .convertShape(VectorOperators.L2I, IntVector.SPECIES_512, 0)
+                                  .reinterpretAsInts()
+                                  .intoMemorySegment(ms, offset, bo);
+                          default -> {
+                              VectorIntrinsics.checkFromIndexSize(offset, length(), ms.byteSize() / 4);
+                              for (int i = 0; i < length(); i++) {
+                                  ms.setAtIndex(ValueLayout.JAVA_INT, i, laneSource(i));
+                              }
```

Comment:
I'm confused with this implementation: isn't `length()` a constant for the specified shuffle vector? I mean it must be `2` for `Double128Shuffle`. So why do we need a switch-case here?

Reply:
This is a product of the templating mechanism that generates these classes. The new code is following the same pattern used for arrays. I can take a closer look at this, but it's following existing style. Refactoring may be a future work piece.

Reply:
The template source is has limitations in what it can conditionally express. It's easier in this case to express the whole switch even though the switch argument is constant.

Reply:
Make sense to me. Thanks for the clarify @igraves @PaulSandoz !